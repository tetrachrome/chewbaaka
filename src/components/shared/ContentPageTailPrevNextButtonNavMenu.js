/**
 * ContentPageTailPrevNextButtonNavMenu.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 01, 2020
 * Updated  : Aug 01, 2020
 */

import React from 'react'

import ContentPageSubsectionTwoColumnContentTemplate from './ContentPageSubsectionTwoColumnContentTemplate'
import ContentPageTailPrevNextButtonNavMenuButton from './ContentPageTailPrevNextButtonNavMenuButton'

export default class ContentPageTailPrevNextButtonNavMenu extends React.Component {

  render() {
    return (
      <div>
        <ContentPageSubsectionTwoColumnContentTemplate
          lhsColumn={{content: this.renderPrevPageNavButton()}}
          rhsColumn={{content: this.renderNextPageNavButton()}}
        />
      </div>
    )
  }

  renderPrevPageNavButton() {
    const prevPage = this.props.pageTailNavMenu.prevPage;

    if ( !prevPage ) {
      return null;
    }

    const label = prevPage.label;
    const href = "/" + label.toLowerCase();

    return (
      <div className="FloatLeft">
        <ContentPageTailPrevNextButtonNavMenuButton
          isRTL={true}
          label={label}
          href={href}
        />
      </div>
    ); 
  }

  renderNextPageNavButton() {
    const nextPage = this.props.pageTailNavMenu.nextPage;

    if ( !nextPage ) {
      return null;
    }

    const label = nextPage.label;
    const href = "/" + label.toLowerCase();
  
    return (
      <div className="FloatRight">
        <ContentPageTailPrevNextButtonNavMenuButton
          isRTL={false}
          label={label}
          href={href}
        />
      </div>
    );
  }
}
