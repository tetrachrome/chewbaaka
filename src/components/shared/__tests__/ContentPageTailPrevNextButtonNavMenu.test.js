/**
 * ContentPageTailPrevNextButtonNavMenu.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Aug 01, 2020
 * Updated  : Aug 13, 2020
 */

import React from 'react'

import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageTailPrevNextButtonNavMenu from '../ContentPageTailPrevNextButtonNavMenu'

import RouterWrapped from '../../../testing/RouterWrapped'

const pageTailNavMenu = {
  prevPage: {
    label: "Biology",
    href: "/biology"
  },
  nextPage: {
    label: "Future",
    href: "/future"
  }
};

test('renders ContentPageTailPrevNextButtonNavMenu component', () => {
  render(
    RouterWrapped(
      <ContentPageTailPrevNextButtonNavMenu
        pageTailNavMenu={pageTailNavMenu}
      />
    )
  );

  const prevPageElement = screen.getByText(pageTailNavMenu.prevPage.label);
  expect(prevPageElement).toBeInTheDocument();

  const nextPageElement = screen.getByText(pageTailNavMenu.nextPage.label);
  expect(nextPageElement).toBeInTheDocument();
});

test('ContentPageTailPrevNextButtonNavMenu component snapshot', () => {
  const tree = renderer
    .create(
      RouterWrapped(
        <ContentPageTailPrevNextButtonNavMenu
          pageTailNavMenu={pageTailNavMenu}
        />
      )
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
