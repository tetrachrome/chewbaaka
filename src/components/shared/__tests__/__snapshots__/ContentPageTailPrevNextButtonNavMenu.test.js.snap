// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageTailPrevNextButtonNavMenu component snapshot 1`] = `
<div
  className="ContentPageSubsectionTwoColumnContentTemplateOuterContainer"
>
  <div
    className="ContentPageSubsectionTwoColumnContentTemplateInnerContainer"
  >
    <div
      className="ContentPageSubsectionTwoColumnContentTemplateColumnContainer"
    >
      <div
        className="ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainer ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainerLhs"
      >
        <h4
          className="ContentPageSubsectionSubtitle"
        />
        <div
          className="FloatLeft"
        >
          <a
            href="/biology"
            onClick={[Function]}
          >
            <button
              className="ui blue massive icon left labeled button"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="arrow circle left icon"
                onClick={[Function]}
              />
              Biology
            </button>
          </a>
        </div>
      </div>
      <div
        className="ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainer ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainerRhs"
      >
        <h4
          className="ContentPageSubsectionSubtitle"
        />
        <div
          className="FloatRight"
        >
          <a
            href="/future"
            onClick={[Function]}
          >
            <button
              className="ui blue massive icon right labeled button"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="arrow circle right icon"
                onClick={[Function]}
              />
              Future
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
